jinseok/model/likeDao.js

```diff
@@ -0,0 +1,53 @@
+const createLike = async (userId, postId) => {
+  try {
+    return dataSource.query(
+      `
+          INSERT INTO likes (
+            user_id, posts_id
+          ) VALUES (
+            ?, ?
+          )
+        `,
+      [userId, postId]
+    );
+  } catch (e) {
+    console.log(e);
+    res.status(400).json({ message: 'error' });
+  }
+};
+
+const getLike = async (userId, postId) => {
+  try {
+    return await dataSource.query(
+      `
+      SELECT user_id, posts_id 
+        FROM likes 
+        WHERE user_id = ? AND posts_id = ?`,
+      [userId, postId]
+    );
```

Comment:
```suggestion
    const [result] = await dataSource.query(`
      SELECT EXISTS (
        SELECT
          id
        FROM likes
        WHERE user_id = ? AND posts_id = ?
      ) isLiked
      `, [userId, postId]
    );
    
    return !!parseInt(result.isLiked)
```
기존에 진석님이 작성해 주신것과 제가 suggestion으로 남겨드린 코드는 겉으로 보기에 큰 차이가 없어보이나 동작 및 효율 적인 부분에서 차이가 있습니다.

기존 쿼리와 비슷하게 예를 들자면 "SELECT id FROM users WHERE email = ?" 쿼리를 실행하여 email 주소가 일치하는 id를 검색합니다. 그러나 이 쿼리는 결과 집합의 크기에 관계없이 항상 레코드를 반환하며, 이는 결과 집합에 행이 없을 때도 마찬가지입니다. 불필요한 추가 작업을 수행하게 되어, 효율성이 떨어질 수 있다는 얘기입니다.

반면 제안드리는 쿼리는 "SELECT id FROM users WHERE email = ?" 쿼리 대신 "SELECT EXISTS" 구문을 사용합니다. EXISTS는 서브쿼리의 결과가 존재하는지 여부를 나타내는 bool 값(0 또는 1)을 반환합니다. 이 경우, 서브쿼리는 실제값을 반환하지 않고 단순히 존재 여부만 판단합니다. 그러므로 불필요한 추가 작업을 수행하지 않으며, 효율성이 높아집니다.

이러한 동작은 특히 데이터베이스 내의 데이터 크기가 커질수록 차이가 더욱 두드러질 수 있습니다.

만약 기존에 등록된 이메일이 있는지 여부만 확인할 것이라면 SELECT EXISTS와 함께 double bang(`!!`)을 활용하면 true, false 값을 직관적이고 효율적이게 반환할 수 있습니다.

"double bang" 연산자는 자바스크립트에서 불리언 값으로 변환하는 방법 중 하나입니다. 이 연산자를 사용하면 truthy 하거나 falsy한 값들을 bool 값으로 바꿔주며 동시에 코드의 가독성을 높일 수 있습니다.